src/utils/retryUntilValid.js

```diff
@@ -1,13 +1,11 @@
-import readLineAsync from "./readLineAsync.js";
-
-const retryUntilValid = async (promptMessage, validateFunc) => {
-  try {
-    const input = await readLineAsync(promptMessage);
-    const validatedInput = validateFunc(input);
-    return validatedInput;
-  } catch (err) {
-    console.log(err);
-    return retryUntilValid(promptMessage, validateFunc);
+const retryUntilValid = async (getInputFunc, onError) => {
+  while (true) {
+    try {
+      const input = await getInputFunc();
+      return input;
+    } catch (err) {
+      await onError(err);
+    }
   }
 };
```

Comment:
PR본문에 적어주신 고민에 대해서

```suggestion
const retryUntilValid = async (func, onError) => {
  while (true) {
    try {
       return await func()
    } catch (e) {
       onError(e);
    }
  }
};
```

단순하게만 생각하면 getInputFunc 라는 인자명과 func라는 인자명의 차이가 될것 같아요. 
이런식의 추상화가 이루어진다면 더이상 input 함수가 아닌, 어떠한 함수라도 괜찮게 되었으니, 조금 더 범용적이게 되었다고 볼 수 있을 것 같아요. 현재 diff에서 이전 버전과 비교해서 차이를 생각해봐도 좋을 것 같아요.

(이 유틸함수는 step2에서 사용하기 까다로워 진 것처럼, 랜덤 유틸보다는 사용할 케이스가 많지는 않구나~ 라고 생각해볼수도 있을것 같아요)